python/pip_install/pip_repository.bzl

```diff
@@ -72,8 +74,38 @@ def _resolve_python_interpreter(rctx):
     python_interpreter = _get_python_interpreter_attr(rctx)
 
     if rctx.attr.python_interpreter_target != None:
-        target = rctx.attr.python_interpreter_target
-        python_interpreter = rctx.path(target)
+        # If we have @@ we have bzlmod so we need to hand Windows differently.
+        if str(Label("//:unused")).startswith("@@"):
+            (os, _) = get_host_os_arch(rctx)
+            # If we have Windows the symlink will not work directly and we need
+            # to run a powershell script.
+            # The powershell script will resolve the Python target and
+            # return that string.
+            if os == WINDOWS_NAME:
+                path = rctx.path(rctx.attr.python_interpreter_target).dirname
```

Comment:
Did you try the `.realpath` I mentioned in chat yet?

`path = rctx.path(rctx.attr.python_interpreter_target).realpath`

This should resolve any symlinks without having to call powershell.